quarkus/extensions/kogito-quarkus-extension-common/kogito-quarkus-common-deployment/src/main/java/org/kie/kogito/quarkus/common/deployment/KogitoQuarkusResourceUtils.java

```diff
@@ -162,12 +162,15 @@ public static void registerResources(Collection<GeneratedFile> generatedFiles,
     }
 
     public static IndexView generateAggregatedIndex(IndexView baseIndex, List<KogitoGeneratedClassesBuildItem> generatedKogitoClasses) {
-        List<IndexView> indexes = new ArrayList<>();
-        indexes.add(baseIndex);
-
-        indexes.addAll(generatedKogitoClasses.stream()
+        return generateAggregatedIndexNew(baseIndex, generatedKogitoClasses.stream()
                 .map(KogitoGeneratedClassesBuildItem::getIndexedClasses)
                 .collect(Collectors.toList()));
+    }
+
+    public static IndexView generateAggregatedIndexNew(IndexView baseIndex, List<IndexView> newIndexViews) {
+        List<IndexView> indexes = new ArrayList<>();
+        indexes.add(baseIndex);
+        indexes.addAll(newIndexViews);
         return CompositeIndex.create(indexes.toArray(new IndexView[0]));
```

Comment:
```suggestion
        return CompositeIndex.create(indexes);
```
[Unless the create(Collection](https://github.com/smallrye/jandex/blob/main/core/src/main/java/org/jboss/jandex/CompositeIndex.java#L49-L51)